evergreen.yml

```diff
@@ -119,10 +119,6 @@ tasks:
       short_name: genny
       install_dir: dist
 
-- name: t_rm_toolchain_dir_static_host
```

Comment:
Seems like https://github.com/mongodb/genny/blob/336a7cca9525b066a3425d7775381668f6fb11e3/evergreen.yml#L221 needs to also be removed. It's probably worth keeping the `tg_compile_static_host` task group for consistency and ease of adding new tasks back.

Reply:
Yep.

I also changed all hosts to run compile in a task group to be consistent with other hosts